src/test/java/test/tesng1046/TestClassSample.java

```diff
@@ -0,0 +1,52 @@
+package test.tesng1046;
+
+import org.testng.IAlterTestName;
+import org.testng.IHookCallBack;
+import org.testng.IHookable;
+import org.testng.ITestResult;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+public class TestClassSample implements IHookable {
+
+    @DataProvider (name = "dp", parallel = true)
+    public Object[][] getTestData() {
+        return new Object[][] {
+            {1}, {2}, {3}, {4}, {5}
+        };
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample1(int num) {
+        assertTrue(num != 0);
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample2(int num) {
+        assertTrue(num != 0);
```

Comment:
Sure, it is not important.
But the idea is just to not let newcomers thinking it is a best practice because testng guys are doing it in their samples.